src/coreclr/interpreter/compiler.cpp

```diff
@@ -2216,7 +2457,365 @@ int InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
+
+                // TODO Special case array ctor / string ctor
+                m_pStackPointer -= numArgs;
+
+                // Allocate callArgs for the call, this + numArgs + terminator
+                int32_t *callArgs = (int32_t*) AllocMemPool((numArgs + 2) * sizeof(int32_t));
+                for (int i = 0; i < numArgs; i++)
+                    callArgs[i + 1] = m_pStackPointer[i].var;
+                callArgs[numArgs + 1] = -1;
+
+                // Push the return value and `this` argument to the ctor
+                InterpType retType = GetInterpType(m_compHnd->asCorInfoType(ctorClass));
+                int32_t vtsize = 0;
+                if (retType == InterpTypeVT)
+                {
+                    vtsize = m_compHnd->getClassSize(ctorClass);
+                    PushTypeVT(ctorClass, vtsize);
+                    PushInterpType(InterpTypeByRef, NULL);
+                }
+                else
+                {
+                    PushInterpType(retType, ctorClass);
+                    PushInterpType(retType, ctorClass);
+                }
+                int32_t dVar = m_pStackPointer[-2].var;
+                int32_t thisVar = m_pStackPointer[-1].var;
+                // Consider this arg as being defined, although newobj defines it
+                AddIns(INTOP_DEF);
+                m_pLastIns->SetDVar(thisVar);
+                callArgs[0] = thisVar;
+
+                if (retType == InterpTypeVT)
+                {
+                    AddIns(INTOP_NEWOBJ_VT);
+                    m_pLastIns->data[1] = (int32_t)ALIGN_UP_TO(vtsize, INTERP_STACK_SLOT_SIZE);
+                }
+                else
+                {
+                    AddIns(INTOP_NEWOBJ);
+                    m_pLastIns->data[1] = GetDataItemIndex(ctorClass);
+                }
+                m_pLastIns->data[0] = GetMethodDataItemIndex(ctorMethod);
+                m_pLastIns->SetSVar(CALL_ARGS_SVAR);
+                m_pLastIns->SetDVar(dVar);
+
+                m_pLastIns->flags |= INTERP_INST_FLAG_CALL;
+                m_pLastIns->info.pCallInfo = (InterpCallInfo*)AllocMemPool0(sizeof(InterpCallInfo));
+                m_pLastIns->info.pCallInfo->pCallArgs = callArgs;
+
+                // Pop this, the result of the newobj still remains on the stack
+                m_pStackPointer--;
+                break;
+            }
+            case CEE_POP:
+                CHECK_STACK(1);
+                AddIns(INTOP_NOP);
+                m_pStackPointer--;
+                m_ip++;
+                break;
+            case CEE_LDFLDA:
+            {
+                CORINFO_RESOLVED_TOKEN resolvedToken;
+                CORINFO_FIELD_INFO fieldInfo;
+                uint32_t token = getU4LittleEndian(m_ip + 1);
+                ResolveToken(token, CORINFO_TOKENKIND_Field, &resolvedToken);
+                m_compHnd->getFieldInfo(&resolvedToken, m_methodHnd, CORINFO_ACCESS_GET, &fieldInfo);
```

Comment:
```suggestion
                m_compHnd->getFieldInfo(&resolvedToken, m_methodHnd, CORINFO_ACCESS_ADDRESS, &fieldInfo);
```
ldflda -> CORINFO_ACCESS_ADDRESS